### PR TITLE
Thorough scouring of docs + doc build infrastructure

### LINKED ISSUE #141: love and care for docstrings, etc.
In working through #135 and making some doc edits, I'm seeing that the docstrings and notebooks are in need a thorough scouring for consistent formatting, grammar & spelling, etc. And along with that an update to the `docs/` infrastructure itself. I'll get to working on that and split it from the work I've already started on in #135.

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* ~~swaps out `conda` for `micromamba` for up to 3x faster environment solves~~
* ~~adds dedicated general and spatial package version printing~~
* adds linting as a pre-req for testing to run
* Resolves #141

### CODE REVIEW

File: mapclassify/classifiers.py [hunk 1]
```diff
@@ -198,7 +201,7 @@ def head_tail_breaks(values, cuts):
     """
     values = np.array(values)
     mean = np.mean(values)
-    if len(cuts) > 0 and cuts[-1] == mean:  # fix floating point issue #117
```
Comment (martinfleis) [inline mapclassify/classifiers.py:201]: Can you leave the comment here? It may not be immediately clear why we do this so the link to #117 will help.

Comment (jGaboardi) [reply mapclassify/classifiers.py:201]: Sure, and I'll update the comment's wording. I deleted because it reads more like a 'to do' than a 'done'.

**Comment by jGaboardi** [comment]
This PR morphed into some monster doc work (see #143). Going to keep this PR going as a DOC [WIP] and open a new, smaller one specifically for the original intent: CI updates.

**Comment by jGaboardi** [comment]
Major points
* docstrings
  * review all docstrings for grammar, spelling, punctuation, clarity etc.
  * standardize docstring formatting
  * add math & code formatting where appropriate
* tutorials
  * review all tutorials and reformat where appropriate
  * clean up cells, code, and viz
  * review all text for grammar, spelling, punctuation, clarity etc.
* doc site and doc build infrastructure
  * update `conf.py` and `build_docs` workflow
  * update `Makefile` to copy over tutorials form `./notebooks` to `./docs/notebooks` before rendering
  * add binder links to tutorials
  * rebuilt doc site locally to ensure everything renders properly

**Comment by jGaboardi** [comment]
@sjsrey Got time for a not-so-quick....? 😨

**Review by sjsrey — APPROVED** [review]
Huge amount of work here. Awesome.

File: requirements_docs.txt [hunk 2]
```diff
@@ -1,5 +1,6 @@
+nbshpinx
```
Comment (QuLogic) [inline requirements_docs.txt:1]: Should be `nbsphinx`, no?

Comment (jGaboardi) [reply requirements_docs.txt:1]: Absolutely yes. Good catch.

Comment (jGaboardi) [reply requirements_docs.txt:1]: xref #174

**Pull request merged**